Examples/CMakeLists.txt

```diff
@@ -12,13 +12,26 @@ endforeach()
 
 if(NOT CMAKE_PROJECT_NAME STREQUAL "SimpleITK")
   find_package(SimpleITK REQUIRED)
-
 endif()
 
 option(SimpleITK_TESTING_NOSHOW "Disable the Show function in SimpleITK testing." ON)
 mark_as_advanced(SimpleITK_TESTING_NOSHOW)
 
-# Add individual cxx executables
+# TODO (Niels Dekker, LKEB, 2022) ON --> OFF
+option(SimpleITK_USE_ELASTIX "Use the Elastix image registration library" ON)
```

Comment:
The CircleCI configuration is now failing when trying to build "Examples" with: 
```
-- Found R: /usr/share/R/include (found version "3.5.2") 
CMake Error at /home/circleci/SimpleITK/Examples/CMakeLists.txt:25 (include):
  include could not find load file:

    /home/circleci/SimpleITK-R/../Elastix-build/ElastixTargets.cmake


-- Configuring incomplete, errors occurred!
```

The loaded SimpleITK configuration should contain the option for the using Elastix. So that this should be a loaded variable and not a run time configuration.

Reply:
@blowekamp Thank you for your encouraging feedback. I think at this point in the CMake code, the Examples project should "ask" to the SimpleITK project if SimpleITK has been built with the `SimpleITK_USE_ELASTIX` option. Right? (Because it that case, it must also link to the Elastix binaries, and it must also build the Elastix examples.)

But then, can you tell me how the Examples project should ask the SimpleITK project if it has been built with `SimpleITK_USE_ELASTIX`?

Reply:
Yes, that is correct.

I believe SimpleITK_USE_ELASTIX should be a configured variable in "SimpleITKConfig.cmake. Additionally find_package should be call for elastix in the config file similar to ITK: https://github.com/SimpleITK/SimpleITK/blob/master/SimpleITKConfig.cmake.in#L63

There are some details to get right if elastics uses modern CMake target export properties or just old cmake variables. SimpleITK is using the modern target approach. IT would be easiest if Elastix also used this approach.

Reply:
This still needs to be done.

Reply:
Please let me know if you need further assistance with this task.

Reply:
I'm not sure about the lib variable warning.

The weak symbol warning is related to Elastix being compiled with the default visibility settings. The following logic may need to be added:
https://github.com/SimpleITK/SimpleITK/blob/master/SuperBuild/External_ITK.cmake#L76-L84

I'm working on a commit for the installed Elastix configuration needed for the example configuration.

Reply:
@blowekamp https://open.cdash.org/viewBuildError.php?type=1&buildid=7827934 (Hosted Agent-Windows) says, before producing "warning CS1668: Invalid search path":

> Microsoft (R) Visual C# 2005 Compiler version 8.00.50727.9031
> for Microsoft (R) Windows (R) 2005 Framework version 2.0.50727
> Copyright (C) Microsoft Corporation 2001-2005. All rights reserved.

Is that intended? 2005 sounds like quite a long time ago now!

Reply:
Yes it does. It looks like it is for just CSharp. I am not too sure about the expect versioning and support from MS. But it's is still included in the recent MSVC, so I assume is has some level of support and patching from MS. Using the old version should enable the library to be compatible with more recent versions for the users. We don't use much C# so no need for newer features for the library. If we hear something from the users, or support is dropped from MSVC then it can be updates as needed.

It is interesting to note that "CS1668" is a C# compiler warning. For some reason I am suspicion that a command line or path is too long or something.

It is just a warning and addressing it can be postponed.

Reply:
@blowekamp [This force-push](https://github.com/SimpleITK/SimpleITK/compare/7d89952b441980d66234b3b10256781e13e8c07c..4d4915c2b0747e703c8895c16f480b7c738d8af1) should ensure that the `SimpleITK_USE_ELASTIX` flag is passed from the main project to the Examples (via SimpleITKConfig.cmake). Please check!